scikeras/wrappers.py

```diff
@@ -386,6 +372,22 @@ def _fit_keras_model(self, X, y, sample_weight, warm_start, epochs, initial_epoc
             ValueError : In case sample_weight != None and the Keras model's
                         `fit` method does not support that parameter.
         """
+        # Make sure model has a loss function
+        if not getattr(self.model_, "loss", None) or (
+            isinstance(self.model_, list)
+            and not any(callable(loss) or isinstance(loss, str) for loss in self.model_)
+        ):
+            raise ValueError(
+                "No valid loss function found."
+                " You must provide a loss function to train."
+                "\n\nTo resolve this issue, do one of the following:"
+                "\n 1. Provide a loss function via the loss parameter."
+                "\n 2. Compile your model with a loss function inside the"
+                " model-building method."
+                "\n\nSee https://www.tensorflow.org/api_docs/python/tf/keras/losses"
+                " for more information on Keras losses."
+            )
```

Comment:
Moving this check to `_fit_keras_model` so that inference/pertained models do not require a loss function.